MyDataHelpsKit/ProviderConnections/ExternalAccountProviders.swift

```diff
@@ -0,0 +1,110 @@
+    ///
+    /// Upon receiving the completion callback, you must present an `SFSafariViewController` to the user using the secure URL in the callback's result object to complete the provider authorization flow.
+    ///
+    /// Upon completion of the connection flow in the Safari view, the participant is sent to the `finalRedirectURL` to indicate that the browser can be dismissed. Your app should intercept this URL via the AppDelegate's `application(_:open:options:)` or `application(_:continue:restorationHandler:)` or the SwiftUI `onOpenURL` modifier, and programmatically dismiss the SFSafariViewController when the URL is opened. Your app can use a specific path in this URL in order to differentiate it from other URLs your app may support.
+    ///
+    /// The user should also be allowed to manually dismiss the Safari view at any time to cancel the authorization or escape if there is an error. In these cases the finalRedirectURL will not be invoked, but you can use `SFSafariViewControllerDelegate` to be notified of manual dismissal if needed.
+    ///
+    /// There are two options for configuring`finalRedirectURL`:
+    /// - [Universal Link](https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html), e.g. `https://my.app/linkprovidercompletion`. The Universal Link domain and path must be fully configured in your app's entitlements file, the apple-app-site-association file hosted at the `my.app` domain, etc.
+    /// - [Custom scheme](https://developer.apple.com/documentation/xcode/defining-a-custom-url-scheme-for-your-app), e.g. `myapp://linkprovidercompletion`. Your app's Info.plist file must register the scheme `myapp` in the URL Types list.
+    ///
+    /// Contact [CareEvolution support](https://developer.rkstudio.careevolution.com/help.html) to have your `finalRedirectURL` added to the list of allowed URLs in RKStudio. If the URL is not in the allow list, this API will produce an error result.
+    /// - Parameters:
+    ///   - provider: The external account provider to connect.
+    ///   - finalRedirectURL: A URL that is configured to open your app via a custom scheme or Universal Link.
+    ///   - completion: Called when the request is complete, with the provider authorization URL and supporting information on success, or an error on failure.
+    func connectExternalAccount(provider: ExternalAccountProvider, finalRedirectURL: URL, completion: @escaping (Result<ExternalAccountAuthorization, MyDataHelpsError>) -> Void) {
+        load(resource: ConnectExternalAccountResource(providerID: provider.id, finalRedirectURL: finalRedirectURL)) {
+            completion($0.map {
+                ExternalAccountAuthorization(provider: provider, authorizationURL: $0, finalRedirectURL: finalRedirectURL)
```

Comment:
Do we need to pass the provider and finalRedirectURL parameters back into the completion? The caller has those values already.

Reply:
Not strictly necessary, I just think it makes things easier for the caller to have everything in a single result object, given the asynchronous flow.